LiveStream/ViewModels/LiveStreamViewModel.swift

```diff
@@ -301,18 +329,106 @@ internal final class LiveStreamViewModel: LiveStreamViewModelType, LiveStreamVie
       replayState
     )
 
-    self.initializeFirebase = configData
-      .filter { event, _ in event.liveNow }
+    let bufferInterval = self.viewDidLoadProperty.signal.flatMap {
+      timer(interval: .milliseconds(500), on: environment.scheduler)
+    }.take(during: Lifetime(self.token))
+
+    let snapshots = self.viewDidLoadProperty.signal
+      .flatMap { [snapshot = self.receivedChatMessageSnapshotProperty.producer] in
+        snapshot
+          .start(on: environment.backgroundQueueScheduler)
+          .skipNil()
+    }
+
+    self.chatMessages = Signal.merge(
```

Comment:
So this signal buffers the chat messages and decodes the arrays of Firebase snapshots into arrays of `LiveStreamChatMessage`.

Feels like it would be useful to try and move this to the `LiveStreamService` so that it can be mocked in the `AppEnvironment`, however seeing as the service is a singleton I'm just wondering how we'd do that whilst keeping this signal clearly `LiveStreamEvent`-specific.